modules/packages/ZMQ.chpl

```diff
@@ -908,23 +904,35 @@ module ZMQ {
 
     // recv, enumerated types
     pragma "no doc"
-    proc recv(type T, flags: int = 0) where isEnumType(T) {
-      return recv(int, flags):T;
+    proc recv(type T, flags: int = 0) throws where isEnumType(T) {
+      return try recv(int, flags):T;
     }
 
     // recv, records (of other supported things)
     pragma "no doc"
-    proc recv(type T, flags: int = 0) where (isRecordType(T) && (!isString(T))) {
+    proc recv(type T, flags: int = 0) throws where (isRecordType(T) &&
+                                                   (!isString(T))) {
       var ret: T;
       on classRef.home {
         var data: T;
-        for param i in 1..numFields(T) do
-          getFieldRef(data,i) = recv(getField(data,i).type);
+        try {
+          for param i in 1..numFields(T) do
+            getFieldRef(data,i) = recv(getField(data,i).type);
+        } catch e: TaskErrors {
+          throw e;
+        }
         ret = data;
       }
       return ret;
     }
 
+    pragma "no doc"
+    proc throw_socket_error(fmtstr: string) throws {
+      var _errno: c_int = errno;
```

Comment:
I tend to prefer it when uses of errno are explicit and at the call site of the place producing the error. For that reason, I'd add errno as an argument to this function and have the callers pass it.